utils/frame/try-runtime/cli/tests/create_snapshot.rs

```diff
@@ -0,0 +1,82 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+#![cfg(unix)]
+#[cfg(feature = "try-runtime")]
+mod tests {
+	use assert_cmd::cargo::cargo_bin;
+	use node_primitives::Hash;
+	use regex::Regex;
+	use std::{
+		path::{Path, PathBuf},
+		process,
+		time::Duration,
+	};
+	use substrate_cli_test_utils as common;
+	use tokio::process::{Child, Command};
+
+	#[tokio::test]
+	async fn create_snapshot_works() {
+		// Build substrate so binaries used in the test use the latest code.
+		common::build_substrate(&["--features=try-runtime"]);
```

Comment:
I now understood why we have to build again. Just very annoying and slow…  
Maybe we can somehow inherit the build args in this function, so that it re-uses the cargo cache.

Reply:
That would be great. It does take quite some time to run all of these tests.

Reply:
Yea that would be a follow up as well; debug that `build_substrate` function and check if we can make it re-use the already existing build artifacts by using the same compile args that were used for the test.